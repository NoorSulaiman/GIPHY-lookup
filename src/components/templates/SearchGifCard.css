#gifcard{
    transition: transform .1s
}

#gifcard:hover{
    transform: scale(1.2);
    z-index: 1000;
}

#gifcard:hover #favIcon{
    display: block;
}

#gifcard:hover #gifTitle{
    display: block;
}

#gifcard:active {
	transform: scale(1.15,1.15) translateY(2px)
}

#favIcon{
    padding-top: 7px;	
	padding-left: 7px;
	position: absolute;
	left: 0;
	top: 0;
    display: none;
    color:rgb(224, 224, 224);
}

.favActive{
    color:rgb(214, 214, 36) !important;
}

@media only screen and (max-width:786px){
    img{
        width:100%;
        max-width: 100%;
        overflow: hidden;
        position: relative;
        margin: 0 auto;
    }
    #gifcard{
        transition: none
    }
    #gifcard:hover{
        transform: none;
        z-index: unset;
    }
    #gifcard:active {
        transform: scale(0.9,0.9) translateY(2px)
    }
    #favIcon{
        padding: 0;
        position:unset;
        margin: 0 auto;
        display: block;
        color:rgb(190, 185, 185);
    }
}