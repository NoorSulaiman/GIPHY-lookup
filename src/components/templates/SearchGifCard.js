import React from 'react';
import PropTypes from 'prop-types';
import { Card, Icon } from 'semantic-ui-react';
import './SearchGifCard.css';

const GifCard = ({ title, imgUrl, toggleFavorite, id, isFave }) => (
    <Card id='gifcard' onClick={() => toggleFavorite({ title, imgUrl, id, isFave })}>
        <img alt={`${title}`} src={imgUrl} />
        {
            isFave ?
                <Icon size='large' id='favIcon' name='favorite' className='favActive' />
                : <Icon size='large' id='favIcon' name='favorite' />
        }
        <Card.Content>
            <Card.Header>{title}</Card.Header>
        </Card.Content>
    </Card >
);



GifCard.propTypes = {
    title: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    toggleFavorite: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired,
    isFave: PropTypes.bool.isRequired,
};

export default GifCard;