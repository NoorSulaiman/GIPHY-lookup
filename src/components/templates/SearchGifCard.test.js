import React from 'react';
import { shallow } from 'enzyme';
import SearchGifCard from './SearchGifCard';

const clickFn = jest.fn();

describe('SearchGifCard', () => {
    it('should render SearchGifCard correctly with given props', () => {
        const component = shallow(<SearchGifCard title={'title'} imgUrl={'url'} id={'id'} toggleFavorite={clickFn} isFave={false} />);
        expect(component).toMatchSnapshot();
        component.find('#gifcard').simulate('click');
        expect(clickFn).toHaveBeenCalled();
        expect(component.find('.favActive').length).toBe(0);
    });

    it('should render the activated favorite icon correctly if isFave true', () => {
        const component = shallow(<SearchGifCard title={'title'} imgUrl={'url'} id={'id'} toggleFavorite={clickFn} isFave />);
        expect(component.find('.favActive').length).toBe(1);
    });
});
