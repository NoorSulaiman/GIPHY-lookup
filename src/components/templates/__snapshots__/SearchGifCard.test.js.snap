// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchGifCard should render SearchGifCard correctly with given props 1`] = `
<Card
  id="gifcard"
  onClick={[Function]}
>
  <img
    alt="title"
    src="url"
  />
  <Icon
    as="i"
    id="favIcon"
    name="favorite"
    size="large"
  />
  <CardContent>
    <CardHeader>
      title
    </CardHeader>
  </CardContent>
</Card>
`;
